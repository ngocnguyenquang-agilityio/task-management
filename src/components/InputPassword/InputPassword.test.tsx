import { render, screen, fireEvent, waitFor } from '@testing-library/react';

import { InputPassword } from '.';

describe('InputPassword component', () => {
  const handleChange = jest.fn();

  test('render the default InputPassword', () => {
    expect(render(<InputPassword onChange={handleChange} />)).toMatchSnapshot();
  });

  test('Test handle change events', () => {
    render(<InputPassword onChange={handleChange} />);
    fireEvent.change(screen.getByTestId('input-password'), {
      target: { value: 'Test input' },
    });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  test('Test show/hide password', async () => {
    const { getByTestId, getByTitle } = render(
      <InputPassword onChange={handleChange} />,
    );

    fireEvent.click(getByTestId('toggle-button'));

    await waitFor(() => {
      expect(getByTitle('hide-password')).toBeInTheDocument();
    });
  });

  test('Test input when disabled', () => {
    const { queryByTestId } = render(
      <InputPassword onChange={handleChange} disabled />,
    );
    expect(queryByTestId('toggle-button')).not.toBeInTheDocument();
  });
});
