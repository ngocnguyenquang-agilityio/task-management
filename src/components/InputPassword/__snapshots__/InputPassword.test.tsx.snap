// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword component render the default InputPassword 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
      >
        <input
          class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-zinc-50 text-zinc-700 dark:bg-zinc-800 dark:text-white p-4 border border-zinc-300 rounded-lg w-full dark:border-gray-700 bg-transparent text-zinc-500 dark:text-white disabled:cursor-not-allowed disabled:opacity-50 focus:outline focus:outline-zinc-300 dark:focus:outline-gray-700 dark:focus:border-gray-700"
          data-testid="input-password"
          placeholder="Search"
          type="password"
          value=""
        />
        <div
          class="absolute inset-y-0 end-0 flex items-center pe-4 cursor-pointer"
          data-testid="toggle-button"
        >
          <svg
            class="h-6 w-6 text-slate-500"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              show-password
            </title>
            <path
              d="M1.18164 12C2.12215 6.87976 6.60812 3 12.0003 3C17.3924 3 21.8784 6.87976 22.8189 12C21.8784 17.1202 17.3924 21 12.0003 21C6.60812 21 2.12215 17.1202 1.18164 12ZM12.0003 17C14.7617 17 17.0003 14.7614 17.0003 12C17.0003 9.23858 14.7617 7 12.0003 7C9.23884 7 7.00026 9.23858 7.00026 12C7.00026 14.7614 9.23884 17 12.0003 17ZM12.0003 15C10.3434 15 9.00026 13.6569 9.00026 12C9.00026 10.3431 10.3434 9 12.0003 9C13.6571 9 15.0003 10.3431 15.0003 12C15.0003 13.6569 13.6571 15 12.0003 15Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
    >
      <input
        class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-zinc-50 text-zinc-700 dark:bg-zinc-800 dark:text-white p-4 border border-zinc-300 rounded-lg w-full dark:border-gray-700 bg-transparent text-zinc-500 dark:text-white disabled:cursor-not-allowed disabled:opacity-50 focus:outline focus:outline-zinc-300 dark:focus:outline-gray-700 dark:focus:border-gray-700"
        data-testid="input-password"
        placeholder="Search"
        type="password"
        value=""
      />
      <div
        class="absolute inset-y-0 end-0 flex items-center pe-4 cursor-pointer"
        data-testid="toggle-button"
      >
        <svg
          class="h-6 w-6 text-slate-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            show-password
          </title>
          <path
            d="M1.18164 12C2.12215 6.87976 6.60812 3 12.0003 3C17.3924 3 21.8784 6.87976 22.8189 12C21.8784 17.1202 17.3924 21 12.0003 21C6.60812 21 2.12215 17.1202 1.18164 12ZM12.0003 17C14.7617 17 17.0003 14.7614 17.0003 12C17.0003 9.23858 14.7617 7 12.0003 7C9.23884 7 7.00026 9.23858 7.00026 12C7.00026 14.7614 9.23884 17 12.0003 17ZM12.0003 15C10.3434 15 9.00026 13.6569 9.00026 12C9.00026 10.3431 10.3434 9 12.0003 9C13.6571 9 15.0003 10.3431 15.0003 12C15.0003 13.6569 13.6571 15 12.0003 15Z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
